packages/pipeline-editor/src/test-utils.tsx

```diff
@@ -322,6 +324,68 @@ export const samplePipeline = {
   schemas: [],
 };
 
+function createPalette(nodes: CustomNodeSpecification[]): PaletteV3 {
+  const palette = {
+    version: "3.0" as "3.0",
+    categories: [
+      {
+        label: "Nodes",
+        image: "",
+        id: "nodes",
+        description: "Nodes",
+        node_types: [] as CategoryDef["node_types"],
+      },
+    ],
+  };
+
+  for (const node of nodes) {
+    palette.categories[0].node_types!.push({
+      id: "",
+      type: "execution_node",
+      inputs: [
+        {
+          id: "inPort",
+          app_data: {
+            ui_data: {
+              cardinality: {
+                min: 0,
+                max: -1,
+              },
+              label: "Input Port",
+            },
+          },
+        },
+      ],
+      outputs: [
+        {
+          id: "outPort",
+          app_data: {
+            ui_data: {
+              cardinality: {
+                min: 0,
+                max: -1,
+              },
+              label: "Output Port",
+            },
+          },
+        },
+      ],
+      parameters: {},
+      app_data: {
+        ui_data: {
+          label: node.label,
+          description: node.description,
+          image: node.image ?? "",
+          x_pos: 0,
+          y_pos: 0,
+        },
+      },
+      ...node,
```

Comment:
I think this is what was causing your typescript issues. `node` has `type` which would override `type: "execution_node",` above